js/utils/tabFunctions.js

```diff
@@ -37,8 +37,42 @@ function getTabDivPair (tabN, tablayoutId) {
     let div_tab = document.createElement("div");
     div_tab.id = `layout${tablayoutId}.file${tabN}`;
     div_tab.className = `tabcontent belongs_to_layout_${tablayoutId}`;
-    div_tab.contentEditable="true";
-    div_tab.innerHTML = `Start typings in file - ${tabN}`;
+    // div_tab.classList.add("editor");
+    // div_tab.contentEditable="true";
+
+    
+    // Create a div for linenumbers with the class "line-numbers"
+    var lineNumbersDiv = document.createElement("div");
+    lineNumbersDiv.classList.add("line-numbers");
+    lineNumbersDiv.innerHTML = '<span></span>';
+    
+    // Create a textarea element
+    var textarea = document.createElement("textarea");
+    textarea.innerHTML = `Start typings in file - ${tabN}`;
+
+    // Append the lineNumbersDiv and textarea to the parentDiv
+    div_tab.appendChild(lineNumbersDiv);
+    div_tab.appendChild(textarea);
+
+    textarea.addEventListener('keyup', event => {

```

Comment:
Event handlers should be moved to tabEventHandler function or a separate function that caters to the functionality being built at the moment - rethink this please. Also, attachment should happen during the hydration stage (pl refer doHydrate function in TabController).

Reply:
@code-master-ajay

Reply:
@code-master-ajay Checkout the Tab and textmass primitive classes that I defined the sketal of. I think we should place this functionality along inside that. What do you think ?

Reply:
@bot2x sure will do.